src/randoop/FailureSet.java

```diff
@@ -6,35 +6,56 @@
 import java.util.Set;
 
 import randoop.operation.MethodCall;
-import randoop.operation.Operation;
 import randoop.sequence.ExecutableSequence;
 import randoop.sequence.MutableSequence;
 import randoop.sequence.MutableVariable;
+import randoop.sequence.Statement;
 import randoop.sequence.Variable;
 
+/**
+ * FailureSet represents a set of failures that occurred in the execution of a particular
+ * {@link ExecutableSequence}. 
+ *
+ */
 public class FailureSet {
   
   private Set<Failure> failures = new LinkedHashSet<Failure>();
   
+  /**
+   * FailureSet.Failure represents an execution failure and holds the
+   * statement and class where the failure occurred.
+   *
+   */
   public static class Failure {
-    public final Operation st;
-    public final Class<?> viocls;
-    public Failure(Operation st, Class<?> viocls) {
-      this.st = st;
-      this.viocls = viocls;
+    
+    /**
+     * statement is the statement in the {@link ExecutableSequence} where the failure occurred.
+     */
+    public final Statement statement;
+    
+    /**
+     * violationClass is the type of the {@link ObjectContract}
+     */
+    public final Class<?> violationClass;
+    
+    public Failure(Statement st, Class<?> viocls) {
+      this.statement = st;
+      this.violationClass = viocls;
     }
+    
     public boolean equals(Object o) {
       if (o == null) return false;
```

Comment:
This isn't your fault because the poor code was alraeady there, but this will crash if called with a non-Failure argument.  The canonical way that most `equals()` methods should start is

    if (! o instanceof Failure) {
      return null;
    }

which doesn't crash and also makes the `o == null` test unnecessary (since `null` in an `instanceof` expression yields `false`).